### ISSUE TITLE
Error Overlay's errors are output in a different order than the compiler

### ISSUE THREAD

**Issue by tankorsmash** [lydell/elm-watch#50]
Thanks for elm-watch, it's been a game changer!

It seems like the errors are being output in a semi-random order, whereas the compiler appears to be consistent in how it orders the errors. It looks like it's got something to do with how elm-watch tries to cache errors maybe.

If I've got errors on lines 82, 95, and 80, both `elm make` and `elm-test --watch`'ll list them in that order, whereas `elm-watch` lists them out of order as lines 80, 82, 95.

#### To reproduce:

Using https://github.com/lydell/elm-watch/tree/main/example-minimal#readme, add the following to `src/Main.elm`:

```elm
myName3 : Int
myName3 = "asdsd"

myName : Int
myName = "asdsd"

myName2 : Int
myName2 = "asdsd"
```

At first, the errors will be presented in the same order as the compiler, but if you fix what was causing the second error, save, then re-introduce the error, it'll appear at the wrong position.

#### Example 

Here's a recording of what I mean. The left is my editor, and the elm compiler output at the bottom, and elm-watch's output on the right. Towards the end, the `myName` error is listed first from the compiler, but second in elm-watch.

https://user-images.githubusercontent.com/1312390/203904042-6140b66a-ba30-42a5-832f-c784a7f8f024.mp4

#### Workaround

Refresh the page, and the errors will appear in the correct order

**Comment by tankorsmash** [comment]
Thanks for taking the time to fix this!

**Comment by lydell** [comment]
Released in v1.1.1.
